starknet/src/space/space.cairo

```diff
@@ -474,9 +474,12 @@ mod Space {
             SyscallResult::Ok(())
         }
 
-        fn post_upgrade_initializer(
-            ref self: ContractState, initialize_calldata: Array<felt252>,
-        ) { // This contract being the first version, we don't expect anyone to upgrade to it.
+        fn post_upgrade_initializer(ref self: ContractState, initialize_calldata: Array<felt252>,) {
+            // This code is left here to indicate to future developers that this
+            // function should be called only once!
+            let state = Reinitializable::unsafe_new_contract_state();
+            Reinitializable::InternalImpl::initialized(@state);
```

Comment:
this should call `initialize` instead right?

Reply:
yeah i didn't know how to convey the idea exactly... if we call `initialize` then the user won't necessarily understand that it should check that it hasn't already been initialized. Also, the work `initialize` may be confused with the already-existing `initialize` function, which should not be called here (it's the whole idea of `post_upgrade_initialize`...)

Reply:
`initialize` internally checks that it has not been initialized already: https://github.com/snapshot-labs/sx-starknet/blob/4f602aa269ef92e279a346fc029fdebcc516a322/starknet/src/utils/reinitializable.cairo#L15C10-L15C10

the full path to the module is provided so dont think confusion should be an issue.

Reply:
If you think calling `initialize` makes it clearer, then let's go for it :)

Reply:
i see what you mean about `initialize` being confusing, ideally we would have used different naming for the `Reinitializable` module. 

But yeah to me `initialize` is the most clear. If the `post_upgrade_initialize` function was empty (eg space v2 added a new function but no state changes needed) then it could be left as it is.

Reply:
ecfc53bb63bebff4c7ec73fc5e8b07f25d9c88aa added a comment to try to make it clear